### PR TITLE
G Suite: Update urls to better support G Suite and Google Workspace

### PR DESCRIPTION
**Pull request by stephanethomas**
This pull request is a follow-up of https://github.com/Automattic/wp-calypso/pull/51302 which updates urls from the `Add New Users` and `Upsell` pages to be able to correctly handle G Suite and Google Workspace accounts:

Add New Users | Upsell
------ | -----
screenshot | screenshot

As a reminder, the `Add New Users` page is used when a user purchases extra licenses but also when they purchase a new account. If the user is purchasing additional licenses for an existing Google Workspace account, we shouldn't add a G Suite product to the shopping cart (and vice-versa). So the idea is to specify the type of product in the url itself to identify clearly the account. The change to the url of the `Upsell` page is just cosmetic though. Here are the former urls:

* http://calypso.localhost:3000/email/example.com/gsuite/new/basic/example.com
* http://calypso.localhost:3000/email/example.com/gsuite/new/starter/example.com
* http://calypso.localhost:3000/email/example.com/gsuite/add-users/example.com
* http://calypso.localhost:3000/domains/add/example.com/google-apps/example.com

The new urls are more descriptive and consistent:

* http://calypso.localhost:3000/email/example.com/gsuite/new/example.com
* http://calypso.localhost:3000/email/example.com/google-workspace/new/example.com
* http://calypso.localhost:3000/email/example.com/gsuite/add-users/example.com
* http://calypso.localhost:3000/email/example.com/google-workspace/add-users/example.com
* http://calypso.localhost:3000/domains/add/example.com/gsuite/example.com
* http://calypso.localhost:3000/domains/add/example.com/google-workspace/example.com

Finally, this pull request:

* Fixes a wrong title displayed when adding new users to a G Suite account
* Fixes all sites view lost when showing `Add New Users` page
* Removes unnecessary feature gates (and check the product slug instead)
* Removes legacy redirects introduced in https://github.com/Automattic/wp-calypso/pull/31703
* Removes legacy redirects introduced in https://github.com/Automattic/wp-calypso/pull/32189

Regarding the latter, my goal was to simplify our code. Moreover, those redirects have been around for at least two years now so it's probably more than time to remove them.

/cc @hacchism @sstabrizi as some Happiness Engineers may reference those urls in their predefined messages.

#### Testing instructions

You will have to add `?flags=-google-workspace-migration` to the url (and reload the page) to be able to purchase G Suite instead of Google Workspace. Here are some flows to check:

1. Purchase new domain with Google Workspace
2. Purchase new domain with G Suite
3. Add Google Workspace to an existing domain
4. Add G Suite to an existing domain
5. Add new users to an existing Google Workspace account
6. Add new users to an existing G Suite account
7. Perform steps #3 to #6 with a domain-only site

There is no need to perform the actual purchase though, just check that the right information is displayed in the checkout. You can also check with an account with more than one site that the sidebar doesn't change when accessing the `Add New Users` page (assuming the view was switched to `All My Sites`).

### CODE REVIEW

File: client/my-sites/email/controller.js [hunk 1]
```diff
@@ -22,24 +20,22 @@ export default {
...
-		);
-	},
-
 	emailManagementNewGSuiteAccount( pageContext, next ) {
 		pageContext.primary = (
 			<CalypsoShoppingCartProvider>
 				<GSuiteAddUsers
-					planType={ pageContext.params.planType }
+					isNewAccount
+					productType={ pageContext.params.productType }
```
Comment (stephanethomas) [inline client/my-sites/email/controller.js:38]: Since we now always pass a product type to the `GSuiteAddUsers` component, we need a way to understand whether it's for a new account, or for extra licenses.

File: client/my-sites/email/gsuite-add-users/index.jsx [hunk 2]
```diff
@@ -81,24 +79,16 @@ class GSuiteAddUsers extends React.Component {
 	}
 
 	handleContinue = () => {
-		const { domains, planType, selectedSite } = this.props;
+		const { domains, productType, selectedSite } = this.props;
```
Comment (daledupreez) [inline client/my-sites/email/gsuite-add-users/index.jsx:82]: Can we not rely on the productSlug that we're already returning in the `domains` object?
```suggestion
		const { domains, selectedSite } = this.props;
		const currentProductSlug = domains[0]?.googleAppsSubscription?.productSlug; // this can definitely be cleaned up!
		const productSlug = currentProductSlug ?? isEnabled( 'google-workspace-migration' ) ?  GOOGLE_WORKSPACE_BUSINESS_STARTER_YEARLY : GSUITE_BASIC_SLUG;
```

File: client/my-sites/controller.js [hunk 3]
```diff
@@ -189,11 +189,16 @@ function isPathAllowedForDomainOnlySite( path, slug, primaryDomain, contextParam
 		emailManagementTitanControlPanelRedirect,
 	];
 
+	// Injects required parameters for some paths as otherwise comparison won't work well
```
Comment (daledupreez) [inline client/my-sites/controller.js]: I am not sure what this comment means. What comparison won't work?

File: client/my-sites/domains/domain-search/index.jsx [hunk 4]
```diff
@@ -13,13 +13,14 @@ import { withShoppingCart } from '@automattic/shopping-cart';
 /**
  * Internal dependencies
  */
+import config from '@automattic/calypso-config';
```
Comment (daledupreez) [inline client/my-sites/domains/domain-search/index.jsx:16]: 1000% nit: is this not technically an external dependency...?

File: client/my-sites/email/email-providers-comparison/index.jsx [hunk 5]
```diff
@@ -70,14 +76,17 @@ class EmailProvidersComparison extends React.Component {
 	};
 
 	goToAddGSuite = () => {
-		const { domain, currentRoute, selectedSiteSlug } = this.props;
+		const { domain, currentRoute, gSuiteProduct, selectedSiteSlug } = this.props;
```
Comment (daledupreez) [inline client/my-sites/email/email-providers-comparison/index.jsx:79]: We should confirm we don't need to make related changes in the new email providers comparison page added in #51307. I think it's just logic cleanup rather than any blocking changes, though.

File: client/my-sites/email/email-providers-comparison/index.jsx [hunk 6]
```diff
@@ -246,13 +255,15 @@ class EmailProvidersComparison extends React.Component {
 	renderGSuiteDetails( className ) {
 		const { currencyCode, gSuiteProduct, translate } = this.props;
 
-		const logo = config.isEnabled( 'google-workspace-migration' )
```
Comment (daledupreez) [inline client/my-sites/email/email-providers-comparison/index.jsx:222]: The logic cleanup here would be good to carry through to the new comparison page from #51307 as well, but I think that can be in a separate PR.

**Review by daledupreez — APPROVED** [review]
LGTM! I worked through all the testing instructions, including some delays waiting for DNS propagation to allow G Suite/Google Workspace purchases.

File: client/my-sites/controller.js (hunk 3 above)
Comment (stephanethomas) [reply client/my-sites/controller.js]: Path comparison. I tried to refine the original comment that was located a few lines below but obviously this wasn't enough. I'll clarify that 🙂.

File: client/my-sites/domains/domain-search/index.jsx (hunk 4 above)
Comment (stephanethomas) [reply client/my-sites/domains/domain-search/index.jsx:16]: Frankly, I don't know. It used to be but now I'm not sure. I've checked a few other files, and sometimes it's internal and sometimes it's not.

File: client/my-sites/email/email-providers-comparison/index.jsx (hunk 6 above)
Comment (stephanethomas) [reply client/my-sites/email/email-providers-comparison/index.jsx:222]: Gotcha, I'll add a card to take care of that.
